lib/ansible/plugins/cache/__init__.py

```diff
@@ -59,7 +59,8 @@ class BaseCacheModule(AnsiblePlugin):
     _display = display
 
     def __init__(self, *args, **kwargs):
-        self._load_name = self.__module__.split('.')[-1]
+        # get everything after ansible.plugins.cache
+        self._load_name = '.'.join(self.__module__.split('.')[3:])
         super(BaseCacheModule, self).__init__()
```

Comment:
i would say the bug is that the plugin inside the collection is not using a correct  name, it needs to match the collection name, we should not have to make every other plugin or base class collection aware

Reply:
This was true for the inventory plugin in the collection. Thanks for noticing that. Cache is a different case since they do not set their own name. Also kind of feeling like I wish inventory plugins did not set NAME: https://github.com/ansible/ansible/pull/56469#discussion_r285124586

I modified the loader instead to set self._load_name early enough for the plugin `__init__`